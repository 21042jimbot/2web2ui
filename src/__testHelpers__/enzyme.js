export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
