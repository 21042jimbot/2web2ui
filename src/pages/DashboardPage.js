import React, { Component } from 'react';
import { connect } from 'react-redux';

export class DashboardPage extends Component {
  renderKeyValue (object, key) {
    return (
      <tr key={key}>
        <td>{key}</td>
        <td>{JSON.stringify(this.props[object][key])}</td>
      </tr>
    );
  }

  render () {
    return (
      <div>
        <h1>My Dashboard</h1>
        <table>
          <thead>
            <tr><th colSpan={2} style={{ borderBottom: '1px solid #ccc', paddingBottom: '5px' }}>Account</th></tr>
            <tr><th style={{ padding: '5px 0' }}>Key</th><th>Value</th></tr>
          </thead>
          <tbody>
            {this.props.account ? Object.keys(this.props.account).map((key) => this.renderKeyValue('account', key)) : null}
          </tbody>
        </table>

        <br />
        <br />

        <table>
          <thead>
            <tr><th colSpan={2} style={{ borderBottom: '1px solid #ccc', paddingBottom: '5px' }}>User</th></tr>
            <tr><th style={{ padding: '5px 0' }}>Key</th><th>Value</th></tr>
          </thead>
          <tbody>
            {this.props.currentUser ? Object.keys(this.props.currentUser).map((key) => this.renderKeyValue('currentUser', key)) : null}
          </tbody>
        </table>
      </div>
    );
  }
}

export default connect(({ account, currentUser }) => ({ account, currentUser }))(DashboardPage);
