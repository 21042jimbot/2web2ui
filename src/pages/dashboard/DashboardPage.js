import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { Page } from '@sparkpost/matchbox';
import { UsageReport } from 'src/components';
import { GettingStartedGuide } from './components/GettingStartedGuide';
import VerifyEmailBanner from 'src/components/verifyEmailBanner/VerifyEmailBanner';
import { FreePlanWarningBanner } from 'src/pages/billing/components/Banners';
import SignupModal from './components/SignupModal';
/* helpers */
import { getAccountUiOptionValue } from 'src/helpers/conditions/account';
/* actions */
import { setAccountOption } from 'src/actions/account';
import SuppressionBanner from './components/SuppressionBanner';

export function DashboardPage(props) {
  const { accountAgeInDays, currentUser, account, checkSuppression } = props;

  useEffect(() => {
    checkSuppression();
  }, [checkSuppression]);

  const displayGuideAndReport = () => {
    const { onboarding: { isGuideAtBottom } = {}, accountAgeInWeeks, hasSuppressions } = props;
    const usageReport = <UsageReport />;
    const gettingStartedGuide = <GettingStartedGuide {...props} />;
    const suppresionBanner = (
      <SuppressionBanner accountAgeInWeeks={accountAgeInWeeks} hasSuppressions={hasSuppressions} />
    );

    if (isGuideAtBottom) {
      return (
        <>
          {usageReport}
          {suppresionBanner}
          {gettingStartedGuide}
        </>
      );
    }
    return (
      <>
        {suppresionBanner}
        {gettingStartedGuide}
        {usageReport}
      </>
    );
  };

  //Shows banner if within 14 days of plan to downgrade

  return (
    <Page title="Dashboard">
      {currentUser.email_verified === false && (
        <VerifyEmailBanner verifying={currentUser.verifyingEmail} />
      )}
      <SignupModal />
      <FreePlanWarningBanner
        account={account}
        accountAgeInDays={accountAgeInDays}
        ageRangeStart={16}
      />
      {displayGuideAndReport()}
    </Page>
  );
}
const mapStateToProps = state => ({
  onboarding: getAccountUiOptionValue('onboarding')(state),
});

export default connect(mapStateToProps, { setAccountOption })(DashboardPage);
