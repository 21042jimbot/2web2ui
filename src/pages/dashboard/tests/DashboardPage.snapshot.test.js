import { shallow } from 'enzyme';
import React from 'react';

import { DashboardPage } from '../DashboardPage';

const props = {
  currentUser: {
    email_verified: true
  }
};

let wrapper;

beforeEach(() => {
  wrapper = shallow(<DashboardPage {...props} />);
});

it('renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly when user is not verified', () => {
  wrapper.setProps({ currentUser: { email_verfied: false }});
  expect(wrapper).toMatchSnapshot();
});
