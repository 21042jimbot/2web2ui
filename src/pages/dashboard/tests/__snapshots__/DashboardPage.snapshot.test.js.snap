// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Dashboard tests should correctly render page when user is not verified 1`] = `
<Page
  empty={Object {}}
  title="Dashboard"
>
  <EmailBanner
    handleResend={[Function]}
    sendingStatus={null}
  />
  <Connect(UsageReport) />
  <Tutorial
    account={
      Object {
        "status": "active",
        "subscription": Object {
          "code": "paid",
        },
      }
    }
    accountAgeInWeeks={0}
    currentUser={
      Object {
        "email_verfied": false,
      }
    }
    hasSuppressions={true}
    listApiKeys={[Function]}
    listSendingDomains={[Function]}
    listSuppressions={[Function]}
  />
</Page>
`;

exports[`Page: Dashboard tests should display upgrade CTA when account is free and active 1`] = `
<Page
  empty={Object {}}
  title="Dashboard"
>
  <Connect(UsageReport) />
  <Tutorial
    account={
      Object {
        "status": "active",
        "subscription": Object {
          "code": "free",
        },
      }
    }
    accountAgeInWeeks={0}
    currentUser={
      Object {
        "email_verified": true,
      }
    }
    hasSuppressions={true}
    listApiKeys={[Function]}
    listSendingDomains={[Function]}
    listSuppressions={[Function]}
  />
</Page>
`;

exports[`Page: Dashboard tests should render import suppression list when 0 suppressions and new account 1`] = `
<Page
  empty={Object {}}
  title="Dashboard"
>
  <Connect(UsageReport) />
  <Banner
    status="default"
    title="Coming from another email service?"
  >
    <p>
      Welcome! Make sure you import your suppression list from your previous provider to avoid sending to people who have previously opted out, consistently bounced, etc. Learn more about migrating from Mailgun, Mandrill, or SendGrid. 
      <a
        href="https://www.sparkpost.com/docs/getting-started/getting-started-sparkpost/#important-coming-from-other-email-services"
        rel="noopener noreferrer"
        target="_blank"
      >
        import your suppressions
      </a>
      .
    </p>
  </Banner>
  <Tutorial
    account={
      Object {
        "status": "active",
        "subscription": Object {
          "code": "paid",
        },
      }
    }
    accountAgeInWeeks={40}
    currentUser={
      Object {
        "email_verified": true,
      }
    }
    hasSuppressions={false}
    listApiKeys={[Function]}
    listSendingDomains={[Function]}
    listSuppressions={[Function]}
  />
</Page>
`;

exports[`Page: Dashboard tests should render page correctly with defaults 1`] = `
<Page
  empty={Object {}}
  title="Dashboard"
>
  <Connect(UsageReport) />
  <Tutorial
    account={
      Object {
        "status": "active",
        "subscription": Object {
          "code": "paid",
        },
      }
    }
    accountAgeInWeeks={0}
    currentUser={
      Object {
        "email_verified": true,
      }
    }
    hasSuppressions={true}
    listApiKeys={[Function]}
    listSendingDomains={[Function]}
    listSuppressions={[Function]}
  />
</Page>
`;
